datalad/core/distributed/clone.py

```diff
@@ -585,15 +585,125 @@ def postclonecfg_ria(ds, props):
+            # have a valid path to origin's config file:
+            from datalad.support.network import URL
+            from datalad.utils import PurePosixPath
+            cfg_path = PurePosixPath(URL(props['giturl']).path) / 'config'
+            op = SSHRemoteIO(props['giturl'])
+            try:
+                # TODO: double-check path
+                response = op.read_file(cfg_path)
+                config_content = response
+            except RIARemoteError as e:
+                lgr.debug("Failed to get config file from source:\n%s",
+                          exc_str(e))
+
+        elif scheme == 'file':
+
+            # TODO: switch the following to proper command abstraction:
+            from datalad.distributed.ora_remote import LocalIO, RIARemoteError
+            op = LocalIO()
+            from datalad.support.network import URL
+            cfg_path = Path(URL(props['giturl']).localpath) / 'config'
+            try:
+                # TODO: double-check path
+                response = op.read_file(cfg_path)
+                config_content = response
+            except RIARemoteError as e:
+                lgr.debug("Failed to get config file from source:\n%s",
+                          exc_str(e))
+        else:
+            lgr.debug("Unknown URL-Scheme in %s. Can handle HTTP, SSH or FILE."
+                      "", props['source'])
+
+        # 3. And read it
+        org_uuid = None
+        if config_content:
+            from datalad.utils import make_tempfile
+            from datalad.cmd import (
+                GitRunner,
+                WitlessRunner,
+                StdOutCapture
+            )
+
+            with make_tempfile(content=config_content) as cfg_file:
+                runner = WitlessRunner(env=GitRunner.get_git_environ_adjusted())
+                try:
+                    result = runner.run(
+                        ['git', 'config', '-f', cfg_file,
+                         'datalad.ora-remote.uuid'],
+                        protocol=StdOutCapture
+                    )
+                    org_uuid = result['stdout'].strip()
+                except CommandError:
+                    # doesn't contain what we are looking for
+                    lgr.debug("Found nothing in source config.")
+
+        # Now, enable it. If annex-init didn't fail to enable it as stored, we
+        # wouldn't end up here, so enable with store URL as suggested by the URL
+        # we cloned from.
+        if org_uuid:
+            srs = ds.repo.get_special_remotes()
+
+            # TODO: - Double-check autoenable value and only do this when true?
+            #       - What if still fails? -> Annex shouldn't change config in
+            #         that case
+
+            new_url = props['source'].split('#')[0]  # we only need the store
+            ds.repo.enable_remote(srs[org_uuid]['name'],
+                                  options=['url={}'.format(new_url)]
+                                  )
```

Comment:
Should protect against `org_uuid` not being a key in `srs`. Or is this impossible? IOW can it be that the store has a remote on file, but the special remote config in the dataset is not there? I think it can happen (git-annex branch not pushed/uptodate, maybe even special remote config removed).

Reply:
Yes, seems possible to happen.